persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/TreatmentMyBatisRepository.java

```diff
@@ -43,8 +63,17 @@
     }
 
     @Override
-    public Set<String> getAllUniqueTreatments(List<String> sampleIds, List<String> studyIds, String key) {
-        return treatmentMapper.getAllUniqueTreatments(sampleIds, studyIds, key);
+    public Set<String> getAllUniqueTreatments(List<String> sampleIds, List<String> studyIds, ClinicalEventKeyCode key) {
+        return treatmentMapper.getAllUniqueTreatments(sampleIds, studyIds, key.getKey())
+            .stream()
+            .flatMap(treatment -> {
+                if (key.isDelimited()) {
+                    return Arrays.stream(treatment.split(key.getDelimiter()));
+                } else {
+                    return Stream.of(treatment);
+                }
+            })
+            .collect(Collectors.toSet());
```

Comment:
I have one question about this. I know `treatmentMapper.getAllUniqueTreatments` can promise give you unique treatments, but how can we know the result is still unique after we do `treatment.split(key.getDelimiter())`?

Reply:
We don't, but `toSet` will enforce uniqueness, so you'll still end up with the correct output.

Reply:
I ignored that, you are right